src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/commands/mobile/button_press_request.h

```diff
@@ -1,21 +1,60 @@
 #ifndef SRC_COMPONENTS_APPLICATION_MANAGER_RPC_PLUGINS_RC_RPC_PLUGIN_INCLUDE_COMMANDS_BUTTON_PRESS_REQUEST_H
 #define SRC_COMPONENTS_APPLICATION_MANAGER_RPC_PLUGINS_RC_RPC_PLUGIN_INCLUDE_COMMANDS_BUTTON_PRESS_REQUEST_H
 
-#include "application_manager/commands/command_request_impl.h"
+#include "rc_rpc_plugin/commands/rc_command_request.h"
 #include "utils/macro.h"
 
 namespace rc_rpc_plugin {
 namespace app_mngr = application_manager;
 
 namespace commands {
-class ButtonPressRequest : public app_mngr::commands::CommandRequestImpl {
+class ButtonPressRequest : public RCCommandRequest {
  public:
-  ButtonPressRequest(const app_mngr::commands::MessageSharedPtr& message,
-                     app_mngr::ApplicationManager& application_manager);
-  bool Init() OVERRIDE;
-  void Run() OVERRIDE;
-  void on_event(const application_manager::event_engine::Event& event) OVERRIDE;
+  ButtonPressRequest(
+      const app_mngr::commands::MessageSharedPtr& message,
+      app_mngr::ApplicationManager& application_manager,
+      rc_rpc_plugin::ResourceAllocationManager& resource_allocation_manager);
+
+  /**
+   * @brief Execute command
+   */
+  void Execute() FINAL;
+
+  /**
+   * @brief AcquireResource Tries to acquire specific resource
+   * @param message Incoming message containg the resource name
+   * @return Acquire result
+   */
+  AcquireResult::eType AcquireResource(
+      const app_mngr::commands::MessageSharedPtr& message) FINAL;
+
+  /**
+   * @brief IsResourceFree check resource state
+   * @param module_type Resource name
+   * @return True if free, otherwise - false
+   */
+  bool IsResourceFree(const std::string& module_type) const FINAL;
```

Comment:
@LitvinenkoIra are yo usure that such methods should be overriten in ButtonPress?

Reply:
@LuxoftAKutsan yes